megatron/training/training.py

```diff
@@ -205,6 +206,16 @@ def is_save_iteration(iteration: int) -> bool:
     Returns:
         True if we should save checkpoint, False otherwise.
     """
+    args = get_args()
+
+    if args.save_iter_patterns is not None:
+        save_iter_patterns = [re.compile(p) for p in args.save_iter_patterns]
+        iter_str = str(iteration)
+        for p in save_iter_patterns:
+            if p.match(iter_str):
```

Comment:
現状のコードの場合、以下のように渡す必要があります。
`--save-iter-patterns '.$' '.0$' '.00$' '.*000$'`

今回想定される使い方の場合、以下の方が適切かと思います。
```suggestion
            if p.fullmatch(iter_str):
```